src/functions.php

```diff
@@ -8,7 +8,7 @@ function resolve($promiseOrValue = null)
         return $promiseOrValue;
     }
 
-    if (method_exists($promiseOrValue, 'then')) {
+    if (is_object($promiseOrValue) && method_exists($promiseOrValue, 'then')) {
```

Comment:
Maybe add a comment for the reasoning behind the `is_object()` check, eg.

```suggestion
    // Check for object first to avoid method_exists() triggering class autoloaders if $promiseOrValue is a string
    if (is_object($promiseOrValue) && method_exists($promiseOrValue, 'then')) {
```